### PR TITLE
ci: overhaul

### PR DESCRIPTION
**Pull request by Thomaash**
- Use Vis Bot GitHub account.
- Build before NPM publish: places the proper version into the files.
- Lint TypeScript.
- Don't trigger release outside of master branch (it didn't release, but took quite a lot of time to just say “not publishing”).

Addresses #53.

### CODE REVIEW

**Comment by Thomaash** [comment]
Hi @mojoaxel, do we have higher quality logo for our new bot? I found 100×100 px in the repos but GitHub shows it at 272×272 px.

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ executors:
     docker:
       - image: circleci/node:12.7
     working_directory: ~/repo
+    environment:
+      GIT_AUTHOR_EMAIL: tom.vycital+vis-bot@gmail.com
```
Comment (mojoaxel) [inline .circleci/config.yml]: :confused: Wouldn't it be better to have a generic email Adresse without your name in it. This way we could give it also tho other maintainers!?

Comment (Thomaash) [reply .circleci/config.yml]: Yeah, we can change it at any time.

**Comment by Thomaash** [comment]
vis-bot@gmail.com is invalid, vis.bot is taken.

**Comment by mojoaxel** [comment]
> [vis-bot@gmail.com](mailto:vis-bot@gmail.com) is invalid, vis.bot is taken.

visjs.bot@gmail.com

**Comment by Thomaash** [comment]
Already waiting for verification SMS for this one.

**Comment by Thomaash** [comment]
visjsbot@gmail.com is ours.

**Pull request merged**